### ISSUE TITLE
Validate certificate vs private_key

### ISSUE THREAD

**Issue by johnnyshields** [SAML-Toolkits/ruby-saml#675]
Currently there is no validation that certificate actually matches private_key. It would be good to add this, because it may cause headaches for users to debug this if for some reason their keys are out-of-sync.

**Comment by pitbulk** [comment]
We can add a method to check if a pair of cert and private_key are related. Then extend [validate_sp_certs_params!](https://github.com/SAML-Toolkits/ruby-saml/blob/master/lib/onelogin/ruby-saml/settings.rb#L314C11-L314C36) method to use it

**Comment by johnnyshields** [comment]
Right, I think this is only supported on more recent versions of OpenSSL. I will check.

**Comment by pitbulk** [comment]
If that's the case, we can simply check whether the OpenSSL method is available and, based on that, execute the extra check. Rather than forcing people to have a specific OpenSSL version.
